core/src/robot/Robot.cc

```diff
@@ -88,6 +91,12 @@ namespace jiminy
             }
         }
 
+        // Reset the transmissions
+        if (!transmissionsHolder_.empty())
```

Comment:
This is wrong. Here you only reset a single transmission. You need to reset ALL of them:
```
for (auto & transmission : transmissionsHolder_)
{
    transmission->reset();
}
```